pkg/common/cluster/healthchecks/pods.go

```diff
@@ -98,3 +99,19 @@ func filterPods(podList *kubev1.PodList, predicates ...PodPredicate) *kubev1.Pod
 	}
 	return filteredPods
 }
+
+// CheckPendingPods looks for a pod that still remains under pending state for a given number of times in a row
```

Comment:
I think this needs to say that you should re-use the podErrorCount parameter in subsequent calls. That's not intuitively obvious (at least to me). One way to reorganize the code to make this clearer would be to make this a method on a stateful type:

```go
type PodErrorTracker struct {
    Counts map[string]int
    MaxPendingPodsThreshold int
}
// CheckPendingPods checks each pod in the provided list for pending state and updates
// the PodErrorTracker accordingly. It returns nil if no pods were pending more than
// their maximum threshold, and errors if a pod ever exceeds its maximum
// pending threshold.
func (p *PodErrorTracker) CheckPendingPods(podList []kubev1.Pod) (error) {
// implementation details here
}
```